Make systemd-networkd default network manager and DHCPv4/DHCPv6 client (#1925)
scaleway/docs-content: dedibox-network/ipv6/how-to/configure-ipv6-linux.mdx

  - You have an account and are logged into the [Dedibox console](https://console.online.net)
  - You have [created](/dedibox/dedicated-servers/how-to/order-a-server) a Dedibox dedicated server
  - You have [requested](/dedibox-network/ipv6/how-to/request-prefix) a /48 IPv6 prefix
</Message>

> `systemd-networkd` is the network manager of [systemd](https://en.wikipedia.org/wiki/Systemd). It is provided natively on most Linux distributions and can be used for the configuration of IPv6 with its DHCPv6 client feature.

Reviewer: ```suggestion
`systemd-networkd` is the network manager of [systemd](https://systemd.io/). It is provided natively on most Linux distributions and can be used for the configuration of IPv6 with its DHCPv6 client feature.
```

Author: Suggestion applied in local git. Link updated.

<Message type="tip">
  When specifying the IP address of your Dedibox, you need to enter it in [CIDR notation](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing). The netmask for the main IPv4 of a Dedibox server is `/24`, the netmask for a failover IPv4 is `/32`. The netmask for the IPv6 is `/64`.
</Message>

> 1. Find the name of your public Internet interface using the `ip link show` command. In this example, the name of the interface is `eno1`. Take a note of it, as it will be required in the next steps.

Reviewer: ```suggestion
1. Run the `ip link show` command to find the name of your public internet interface. Make sure to store it, as it will be required in the next steps.
    <Message type="note">
      In this example, the name of the interface is `eno1`.
    </Message>
```

Author: Suggestion applied in local git but I would advocate for "Internet" to remain capitalized.

> 2. To configure `systemd-networkd`, start by creating a backup of the current network configuration:

Reviewer: ```suggestion
2. Run the following command to create a backup of the current network configuration and configure `systemd-networkd`:
```

Author: Suggestion applied in my local git.

    ```
    sudo mv /etc/systemd/network/eno1.network /etc/systemd/network/eno1.network.bkp
    ```

> 3. Enable `systemd-networkd` by using the systemctl command. Do not start the service yet, as the server is currently still running on the old configuration and may become unreachable if systemd-networkd is started at this step.

Reviewer: ```suggestion
3. Run the following command to enable `systemd-networkd`. 
    <Message type="important">
      Do not start the service yet, as the server is still running on the old configuration and may become unreachable if `systemd-networkd` is started at this stage.
    </Message>
```

    ```
    systemctl enable systemd-networkd.service
    ```
4. Create a new file called `<interface_name>.network` and open it in a text editor
    ```
    sudo nano /etc/systemd/network/eno1.network
    ```

> 5. Copy the following content in the file, edit it towards your configuration before saving the file and exiting the text editor.

Reviewer: ```suggestion
5. Open the file in a text editor and paste the following content. Make sure to replace the values in the example with your own and save them before exiting the text editor. 
```

Author: Suggestion applied in local git.

    ```
    [Match]
    #The name must correspond to your interface name
    Name=eno1

    [Network]
    DHCP=ipv6
    IPv6AcceptRA=yes
    # IPv4 of the Dedibox, change netmask to /32 if failover IPv4
    Address=<ipv4_of_your_dedibox>/24
    # IPv6 of the Dedibox
    Address=2001:bc8:xxxx:xxx::x/64

    [Route]
    # The IPv4 gateway is either your Dedibox IP ending on .1 (i.e. x.y.z.1) for the physical host or 62.210.0.1 for any VM running on a Dedibox
    Gateway=<gateway_ipv4>
    GatewayOnlink=yes

    [DHCP]
    ClientIdentifier=duid-only
    DUIDType=link-layer
    # Scaleway Dedibox DUID without the DUID-LL "00:03" at the beginning
    DUIDRawData=<Online by Scaleway DUID>

    [IPv6AcceptRA]
    DHCPv6Client=always
    UseAddress=no
    WithoutRA=solicit
    UseOnLinkPrefix=False
    UseAutonomousPrefix=False
    ```

> 10. Start `systemd-networkd` to activate the new configuration.

Reviewer: ```suggestion
10. Run the following command to activate the new configuration.
```

Author: Suggestion applied in local git.
